internal/controller/supervisorconfig/upstreamwatcher/oidc_upstream_watcher.go

```diff
@@ -46,10 +46,10 @@ const (
 	clientSecretDataKey = "clientSecret"
 
 	// Constants related to the OIDC provider discovery cache. These do not affect the cache of JWKS.
-	validatorCacheTTL = 15 * time.Minute
+	oidcValidatorCacheTTL = 15 * time.Minute
```

Comment:
I think many of these small rename changes would be unnecessary if we renamed this original `upstreamwatcher` package to `upstreamoidcwatcher` and added the other new code as a separate `upstreamldapwatcher` package.

Reply:
Is it worth doing that in light of the shared code that would then need to live in yet a third package? I didn't think so, but if y'all would prefer it then I can move it.

Reply:
> Is it worth doing that ...

Yes, please.

Reply:
Actually needed 4 packages. I didn't like splitting one nicely self-contained package into four, but here it is: 1ae3c6a1ad91080d49ec0c45d2960effea63e12d